File: Benchmarking/src/gen_12_7.cpp
  #include <vector>
  #include <iostream>
  #include <chrono>
  struct s0 {
    std::vector<std::vector<double>> p0;
    std::vector<std::vector<double>> p1;
    s0(std::vector<std::vector<double>> v0, std::vector<std::vector<double>> v1): p0(v0), p1(v1) { }
  };
  __attribute__((noinline))
  double f0(const std::vector<std::vector<s0>> &v0, const double &v1) {
    double v5 = v1;
    const std::vector<s0> v8 = v0[1];
    const s0 v3 = v8[0];
    const std::vector<std::vector<double>> v9 = v3.p0;
    const std::vector<double> v4 = v9[0];
    v5 = v1;
    v5 = v1;
    const double v27 = v5 + v1;
    const double v53 = v4[0];
    double v83 = v53;
    const double v98 = v83 / v27;
    return v98;
  }
  int main() {
    std::vector<std::vector<s0>> v0({ { { { { 0.0 } }, { { 1.0 } } } }, { { { { 2.0 } }, { { 3.0 } } } } });
    double v1(4.0);
    auto start = std::chrono::high_resolution_clock::now();
    double result;
    for (int i = 0; i < 1000; i ++) {
      result = f0(v0, v1);
    }
    auto end = std::chrono::high_resolution_clock::now();
    double value = *((double*) &result);
    std::cout << value << "\n";
    std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(end-start).count();
    std::cout << "\n";
    return 0;
  }

File: Benchmarking/src/gen_12_7.scala
import java.lang.System.nanoTime
import scala.collection.immutable.Vector
object Gen extends App {
  case class s0 (
    p0: Vector[Vector[Double]],
    p1: Vector[Vector[Double]]
  )
  @noinline
  def f0(v0: Vector[Vector[s0]], v1: Double): Double = {
    var v5: Double = v1
    val v8: Vector[s0] = v0(1)
    val v3: s0 = v8(0)
    val v9: Vector[Vector[Double]] = v3.p0
    val v4: Vector[Double] = v9(0)
    v5 = v1
    v5 = v1
    val v27: Double = v5 + v1
    val v53: Double = v4(0)
    var v83: Double = v53
    val v98: Double = v83 / v27
    v98
  }
  def benchmark(): Unit = {
    val v0: Vector[Vector[s0]] = Vector(Vector(s0(Vector(Vector(0.0)), Vector(Vector(1.0)))), Vector(s0(Vector(Vector(2.0)), Vector(Vector(3.0)))))
    val v1: Double = 4.0
    val start = nanoTime()
    var result: Double = 4.0
    (1 to 1000).foreach { _ =>
      result = f0(v0, v1)
    }
    val end = nanoTime()
    println(result)
    println(end - start)
  }
  benchmark()
}